components/esp_dns/esp_dns_doh.c

```diff
@@ -0,0 +1,245 @@
+/*
+ * SPDX-FileCopyrightText: 2025 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include "freertos/FreeRTOS.h"
+#include "esp_event.h"
+#include "esp_log.h"
+#include "nvs_flash.h"
+#include "esp_tls.h"
+#include "sdkconfig.h"
+#include "lwip/prot/dns.h"
+#include "lwip/api.h"
+#include "lwip/opt.h"
+#include "lwip/dns.h"
+#include "lwip_default_hooks.h"
+#include "esp_http_client.h"
+#include "esp_dns_utils.h"
+#include "esp_dns_doh.h"
+
+#define TAG "ESP_DNS_DOH"
+
+#define SERVER_URL_MAX_SZ 256
+
+int init_doh_dns(esp_dns_handle_t handle)
+{
+    ESP_LOGI(TAG, "Initializing DNS over HTTPS");
+
+    return 0;
+}
+
+int cleanup_doh_dns(esp_dns_handle_t handle)
+{
+    ESP_LOGI(TAG, "Cleaning up DNS over HTTPS");
+
+    return 0;
+}
```

Comment:
No need for extra initialization for now?

Reply:
Common init and cleanup functions are replaced by init and cleanup function of individual transport type.